### ISSUE TITLE
wfdb.dl_database('afdb', './afdb') fails with TypeError

### ISSUE THREAD

**Issue by kbrose** [MIT-LCP/wfdb-python#118]
```python
>>> import wfdb
>>> wfdb.dl_database('afdb', './afdb')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "...\lib\site-packages\wfdb\io\record.py", line 1394, in dl_database
    for file in record.file_name:
TypeError: 'NoneType' object is not iterable```

**Comment by kbrose** [comment]
If I do this diff, then it fails with a different error:

```diff
diff --git a/wfdb/io/record.py b/wfdb/io/record.py
index dd2ae50..3d92fc5 100644
--- a/wfdb/io/record.py
+++ b/wfdb/io/record.py
@@ -1391,7 +1391,7 @@ def dl_database(db_dir, dl_dir, records='all', annotators='all',
             # Single segment record
             if isinstance(record, Record):
                 # Add all dat files of the segment
-                for file in record.file_name:
+                for file in (record.file_name if record.file_name else []):
                     allfiles.append(posixpath.join(dirname, file))

             # Multi segment record
```

then I get this error message repeated infinitely as far as I can tell:

<details><summary>long traceback - click to expand</summary><p>

```
Downloading files...
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "...\lib\multiprocessing\spawn.py", line 105, in spawn_main
    exitcode = _main(fd)
  File "...\lib\multiprocessing\spawn.py", line 114, in _main
    prepare(preparation_data)
  File "...\lib\multiprocessing\spawn.py", line 225, in prepare
    _fixup_main_from_path(data['init_main_from_path'])
  File "...\lib\multiprocessing\spawn.py", line 277, in _fixup_main_from_path
    run_name="__mp_main__")
  File "...\lib\runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "...\lib\runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "...\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\kevin.rose\Documents\main_code_folder\tdr-pb\py\download.py", line 6, in <module>
    wfdb.io.dl_database(sys.argv[1], Path('.') / sys.argv[1])
  File "...\lib\site-packages\wfdb\io\record.py", line 1430, in dl_database
    pool = multiprocessing.Pool(processes=2)
  File "...\lib\multiprocessing\context.py", line 119, in Pool
    context=self.get_context())
  File "...\lib\multiprocessing\pool.py", line 174, in __init__
    self._repopulate_pool()
  File "...\lib\multiprocessing\pool.py", line 239, in _repopulate_pool
    w.start()
  File "...\lib\multiprocessing\process.py", line 105, in start
    self._popen = self._Popen(self)
  File "...\lib\multiprocessing\context.py", line 322, in _Popen
    return Popen(process_obj)
  File "...\lib\multiprocessing\popen_spawn_win32.py", line 33, in __init__
    prep_data = spawn.get_preparation_data(process_obj._name)
  File "...\lib\multiprocessing\spawn.py", line 143, in get_preparation_data
    _check_not_importing_main()
  File "...\lib\multiprocessing\spawn.py", line 136, in _check_not_importing_main
    is not going to be frozen to produce an executable.''')
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
````

</p></details>

-------------------

If I turn off multiprocessing with the following diff:

```diff
diff --git a/wfdb/io/record.py b/wfdb/io/record.py
index dd2ae50..7eb1b03 100644
--- a/wfdb/io/record.py
+++ b/wfdb/io/record.py
@@ -1427,8 +1427,7 @@ def dl_database(db_dir, dl_dir, records='all', annotators='all',
     print('Downloading files...')
     # Create multiple processes to download files.
     # Limit to 2 connections to avoid overloading the server
-    pool = multiprocessing.Pool(processes=2)
-    pool.map(download.dl_pb_file, dlinputs)
+    [download.dl_pb_file(dlinput) for dlinput in dlinputs]
     print('Finished downloading files')

     return
```

then everything appears to download OK.

**Comment by kbrose** [comment]
Closed with merge of #119.
